### DISCUSSION TITLE
When I mark a column As Primary, I have the option to toggle As Unique. Isn't a primary key unique by definition? Shouldn't the Unique toggle be removed when Primary is toggled on?

### DISCUSSION THREAD

**Discussion by OrhanTozan** [supabase/supabase#20939] [Questions] [answered]
When I mark a column As Primary, I have the option to toggle As Unique. Isn't a primary key unique by definition? Shouldn't the Unique toggle be removed when Primary is toggled on?

**Comment by GaryAustin1** [comment, marked as answer]
You could have a composite primary key with multiple columns that each are not unique is my guess is why it is there.

**Reply by OrhanTozan** [reply]
I guess an improvement would be to only show the Is Unique toggle when there is more than 1 column marked as primary key.
